lib/services/local/headlessauthn_watcher_test.go

```diff
@@ -26,171 +26,324 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
 	"github.com/gravitational/teleport/lib/services"
 	"github.com/gravitational/teleport/lib/services/local"
 )
 
 func TestHeadlessAuthenticationWatcher(t *testing.T) {
+	t.Parallel()
 	ctx := context.Background()
+	pubUUID := services.NewHeadlessAuthenticationID([]byte(sshPubKey))
 
-	t.Parallel()
-	identity := newIdentityService(t, clockwork.NewFakeClock())
+	type testSuite struct {
```

Comment:
Maybe?

```suggestion
	type testEnv struct {
```

I expect a "test suite" to be a collection of test cases, but this looks more like the environment needed to run a test.